packages/react-dev-utils/webpackHotDevClient.js

```diff
@@ -140,19 +140,15 @@ function handleWarnings(warnings) {
     }
   }
 
+  printWarnings();
+
   // Attempt to apply hot updates or reload.
   if (isHotUpdate) {
     tryApplyUpdates(function onSuccessfulHotUpdate() {
-      // Only print warnings if we aren't refreshing the page.
```

Comment:
This is the only thing I'm on the fence about, what's the reason for this change? Can you please add the reasoning in the code?

Reply:
The problem is that we need to print warnings even when there is not update in code. 
Without this change, https://github.com/facebook/create-react-app/blob/68e2f50dacec6205c42a108b6eecfc7f6d99e459/packages/react-dev-utils/webpackHotDevClient.js#L235-L237 will cancel it as `__webpack_hash__` didn't change.